### PR TITLE
fix #5397：index out of range [0] with length 0 during bin()

### LINKED ISSUE #5397: [Bug]:  panic runtime error: index out of range [0] with length 0 during bin()
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

mysql> select bin();
ERROR 20101 (HY000): internal error: panic runtime error: index out of range [0] with length 0: 
runtime.goPanicIndex
	/usr/local/go/src/runtime/panic.go:113
github.com/matrixorigin/matrixone/pkg/sql/plan/function/builtin/unary.generalBin[...]
	/var/root/matrixone/pkg/sql/plan/function/builtin/unary/bin.go:41
github.com/matrixorigin/matrixone/pkg/sql/plan/function/builtin/unary.Bin[...]
	/var/root/matrixone/pkg/sql/plan/function/builtin/unary/bin.go:27
github.com/matrixorigin/matrixone/pkg/sql/plan/function.Function.VecFn
	/va

### Expected Behavior

_No response_

### Steps to Reproduce

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by YANGGMM**
## What type of PR is this?

- [ ] API-change
- [x] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #5397 

## What this PR does / why we need it:
index out of range [0] with length 0 during bin()

### CODE REVIEW

File: pkg/sql/plan/base_binder.go [hunk 1]
```diff
@@ -874,6 +883,9 @@ func bindFuncExprImplByPlanExpr(name string, args []*Expr) (*plan.Expr, error) {
 		}
 	case "like":
 		// sql 'select * from t where col like ?'  the ? Expr's type will be T_any
+		if len(args) == 0 {
```
Comment (ouyuanning) [inline pkg/sql/plan/base_binder.go:886]: you can change to :  if len(args) != 2

Comment (YANGGMM) [reply pkg/sql/plan/base_binder.go:886]: ok

Comment (ouyuanning) [reply pkg/sql/plan/base_binder.go:886]: and add some bvt test

Comment (YANGGMM) [reply pkg/sql/plan/base_binder.go]: ok

**Review by aunjgr — APPROVED** [review]

**Review by aressu1985 — APPROVED** [review]

**Review by ouyuanning — APPROVED** [review]

**Pull request merged**